base/docs/Docs.jl

```diff
@@ -263,11 +263,11 @@ function funcdoc(meta, def, def′, name)
     end
 end
 
-function typedoc(meta, def, name)
+function typedoc(meta, def, def′, name)
     quote
         @init
         $(esc(def))
-        doc!($(esc(name)), $(mdify(meta)), $(field_meta(unblock(def))))
+        doc!($(esc(name)), $(mdify(meta)), $(field_meta(unblock(def′))))
```

Comment:
Oh and you still want to get the metadata out of the definition, of course. Fair enough then.